server/src/main/java/org/candlepin/resource/OwnerResource.java

```diff
@@ -1794,30 +1409,23 @@ public AsyncJobStatusDTO refreshPools(
             .setOwner(owner)
             .setLazyRegeneration(lazyRegen);
 
-        AsyncJobStatus job = this.jobManager.queueJob(config);
+        AsyncJobStatus job;
+
+        try {
+            job = this.jobManager.queueJob(config);
+        }
+        catch (JobException e) {
+            String errmsg = this.i18n.tr("An unexpected exception occurred while scheduling job \"{0}\"",
+                config.getJobKey());
+            log.error(errmsg, e);
+            throw new IseException(errmsg, e);
```

Comment:
Our mappers already handle converting exceptions to the generic "internal server error" if they're not caught. If we don't have explicit error handling to perform, or some other behavior to improve the output for clients. Additionally, the declaration and translation of the job can only happen if the queuing of the job succeeds, so there is no reason to declare the job or perform translation outside the try/catch block.

Is there some behavior change with OpenAPI that I'm not aware of? What's the intent here?

Reply:
As already discussed in PR: https://github.com/candlepin/candlepin/pull/2692#discussion_r421507021. We need to explicitly throw an exception because the auto-generated method does not throws an exception.